File: linkedlist/203.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* removeElements(struct ListNode* head, int val) {
    struct ListNode *cur, *pre;
    struct ListNode *dummy = (struct ListNode*)malloc(sizeof(struct ListNode));
    dummy->next = head;
    dummy->val = -9527;
    cur = head;
    pre = dummy;
    while(cur)
    {
        if(cur->val == val)
        {
            pre->next = cur->next;
            cur = cur->next;
        }
        else
        {
            pre = pre->next;    
            cur = cur->next;
        }
    }
    return dummy->next;
}

File: linkedlist/203.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func removeElements(head *ListNode, val int) *ListNode {
    dummy := &ListNode{0, nil}
    dummy.Next = head
    head = dummy
    for dummy.Next != nil {
        if dummy.Next.Val == val {
            dummy.Next = dummy.Next.Next
        }else {
            dummy = dummy.Next
        }
    }
    return head.Next
}

/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func removeElements(head *ListNode, val int) *ListNode {
    dummy := &ListNode{-1, nil}
    dummy.Next = head
    head = dummy
    for head != nil && head.Next != nil {
        if head.Next.Val == val {
            head.Next = head.Next.Next
        }else {
            head = head.Next    
        }
    }
    return dummy.Next
}
